src/services/user.service.js

```diff
@@ -171,53 +171,82 @@ module.exports = {
   },
 
   /**
-   * Creates a new post of type 'user'.
+   * Creates a new post of type 'user' or 'shared' in case that the
+   * user shares another user post.
    * @param {int} userId 
    * @param {Object} post An object with:
    * - content: string.
    * - image: Object. An object with:
    *   - path: Path of image in the local files.
+   * @param {number} referencedPostId
    */
-  createPost: async function(userId, post) {    
-    // If the user doesn't send any data.
-    if (!post.content && !post.image) {
-      return null
-    }
-
-    let result = {}
+  createPost: async function(userId, post, referencedPostId = null) {
+    let postData = {}
 
     if (post.content) {
-      result.content = post.content;
+      postData.content = post.content;
     }
 
-    if (post.image) {
+    if (post.image && !referencedPostId) {
       try {
         // The image is uploaded to cloudinary
         const resultUploadImage = await cloudinary.uploader.upload(post.image.path)
-        result.img_src = resultUploadImage.secure_url
-        result.cloudinary_id = resultUploadImage.public_id
+        postData.img_src = resultUploadImage.secure_url
+        postData.cloudinary_id = resultUploadImage.public_id
       } catch (err) {
         err.file = __filename
         err.func = 'createPost'
-        err.cloudinary_id = result.cloudinary_id
+        err.cloudinary_id = postData.cloudinary_id
         throw err
       } finally {
          // The local files are deleted.
         fs.unlinkSync(post.image.path)
       }
     }
 
-    let args = [userId, result.content || '', result.img_src || '', result.cloudinary_id || '', 'user']
-    const query = `INSERT INTO posts (user_id, content, img_src, cloudinary_id, post_type) VALUES (?, ?, ?, ?, ?);`
+    let args = []
+    let query = ''
+    if (referencedPostId) {
+      try {
+        // Verify is the referenced post id is a shared post, if is true, it is 
+        // necessary extract the id of the original post.
+        const q = `select post_type, referenced_post_id from posts where id = ?;`
+        let referencedPost = await mariadb.query(q, [referencedPostId])
+        referencedPost = referencedPost[0]
+        if (referencedPost.post_type === 'shared') {
+          referencedPostId = referencedPost.referenced_post_id
+        }
```

Comment:
The new way to detect if it is a shared publication is verifying if the field `referenced_post_id` has an integer positive value.